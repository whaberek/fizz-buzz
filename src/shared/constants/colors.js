const colors = {
  black: '#000000',
  green: '#4B8B3B',
  blue: '#2271B3',
};

export { colors };
