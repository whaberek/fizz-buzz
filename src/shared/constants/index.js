export { colors } from './colors';
export { texts } from './texts';
export { validateConstants } from './validateConstants';
