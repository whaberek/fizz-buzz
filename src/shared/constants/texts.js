const texts = {
  fizz: 'fizz',
  buzz: 'buzz',
  wizz: 'wizz',
  wuzz: 'wuzz',
  submit: 'Submit',
  next: 'Next',
  previous: 'Previous',
};

export { texts };
