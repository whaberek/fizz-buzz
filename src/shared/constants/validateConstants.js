const validateConstants = {
  INTEGER: 'Filled value must be an integer.',
  TOO_SMALL: 'Filled value is too small.',
  TOO_HIGH: 'Filled value is too high.',
};

export { validateConstants };
