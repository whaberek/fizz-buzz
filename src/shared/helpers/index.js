export { checkDay } from './checkDay'
export { modulo } from './modulo'
