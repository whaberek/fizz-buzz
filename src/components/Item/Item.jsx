import styled from 'styled-components';

import { colors } from 'shared/constants';

const Item = styled.span`
  color: ${({ color }) => (color ? color : colors.black)}
`;

export { Item };
