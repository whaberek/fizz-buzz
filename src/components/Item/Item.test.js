import React from 'react';
import { shallow } from 'enzyme';

import { Item } from './Item';

import { colors } from 'shared/constants';

describe('Item', () => {
  it('should render layout with default color', () => {
    const wrapper = shallow(<Item />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render layout with color passed in props', () => {
    const wrapper = shallow(<Item color={colors.green} />);
    expect(wrapper).toMatchSnapshot();
  });
});
