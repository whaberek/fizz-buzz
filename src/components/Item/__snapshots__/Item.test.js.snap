// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item should render layout with color passed in props 1`] = `
<StyledComponent
  color="#4B8B3B"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  color: ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Item should render layout with default color 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": false,
        "rules": Array [
          "
  color: ",
          [Function],
          "
",
        ],
      },
      "displayName": "styled.span",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "span",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
